app/controllers/contact/govuk/accessible_format_requests_controller.rb

```diff
@@ -83,4 +80,18 @@ def display_optional_question?
     trigger_value = current_question[:"trigger-value"]
     params[trigger_key] == trigger_value
   end
+
+  def single_keys
+    questions.map { |question| :"#{question["key"]}" }
+  end
+
+  def multiple_keys
+    multi_key_questions = questions.select do |question|
+      question["type"] == "multi-text"
+    end
+
+    multi_key_questions.map do |question|
+      question["options"].map { |op| :"#{op["value"]}" }
+    end
```

Comment:
Rather than having to differentiate between 'single key' questions and 'multi key questions', I think you could standardise your YAML so that they all have the same configuration. Example:

```yaml
questions:
  - group: set-format
    title: Which accessible format would you like for
    caption: Please choose one
    optional: false
    inputs:
      - type: radio
        key: format
        options:
          - text: Audio format
            value: format-audio
          - text: Braille
            value: format-braile
          - text: British Sign Language
            value: format-bsl
          - text: Large print version
            value: format-large-print
          - text: Not seeing what you're looking for?
            value: format-other
  - group: custom-format
    title: What accessible format do you require?
    caption: Please enter your details here
    optional: true
    trigger-key: set-format
    trigger-value: format-other
    inputs:
      - type: text
        key: custom_format
  - group: contact-details
      title: In order for us to reach you about your request
      caption: Enter your contact details
      optional: false
      inputs:
        - type: text
          label: First name
          key: first_name
        - type: text
          label: Second name
          key: second_name
        - type: text
          label: Email address
          key: email
```

Then your `permitted_params` would be something like `questions.map(&:inputs).pluck(:key).flatten`. But you may want to consider combining the input key with the 'group' to avoid collisions, e.g. `first_name` => `contact-details::first_name`.